Automattic/jetpack
# Quick Share Buttons | Change Twitter icon to X

## Proposed changes:

* Add `XIcon` to `js-packages/components/components/icons`
* Replace Twitter with X in text strings in Quick Share buttons
* Fix the layout and styling of the Quick Share buttons

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:

* Boot up the PR and run `jetpack build plugins/jetpack`
* Create a new post and hit publish or for an existing post, run this in browser console
    - `wp.data.dispatch('core/edit-post').togglePublishSidebar();`
* On the post publish panel, confirm that you see the <s>Twitter</s> X icon
* Confirm that it works as expected
* Open Jetpack sidebar
* Under "Share this post", click on the Quick Share icon
* Confirm that you see the <s>Twitter</s> X icon and the updated text.
* Click on "Learn more..." link
* Confirm the same for the modal

| Before | After |
|--------|--------|
| <img width="283" alt="Screenshot 2023-10-03 at 10 44 13 AM" src="https://github.com/Automattic/jetpack/assets/18226415/5f5bd554-befe-4cf9-933a-70b41039939f"> | <img width="283" alt="Screenshot 2023-10-03 at 10 42 52 AM" src="https://github.com/Automattic/jetpack/assets/18226415/ff09ed5a-522d-4ca5-be27-ebb976ac1b81"> |
| <img width="356" alt="Screenshot 2023-10-03 at 10 44 54 AM" src="https://github.com/Automattic/jetpack/assets/18226415/ef67e07c-778d-4608-8033-4ec42e7673fb"> | <img width="355" alt="Screenshot 2023-10-03 at 10 43 24 AM" src="https://github.com/Automattic/jetpack/assets/18226415/ff33b4fd-ab50-48c2-a577-636bf58dee4e"> |
| <img width="907" alt="Screenshot 2023-10-03 at 10 44 39 AM" src="https://github.com/Automattic/jetpack/assets/18226415/17a750fe-4f13-4300-9ddb-02767493bbe2"> | <img width="908" alt="Screenshot 2023-10-03 at 10 43 06 AM" src="https://github.com/Automattic/jetpack/assets/18226415/26cb5244-0a39-40dc-bd90-bf5a5f2d06f6"> | 


* Test the connections screen. You can change the new function to return `x` every time, then it should show the icons.

![image](https://github.com/Automattic/jetpack/assets/36671565/b7651260-58be-4dcf-81ec-f61bcead423e)
